Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1042,8 +1043,10 @@ BoardInit (
 
     break;
   case EndOfFirmware:
+    ClearSmi ();
     if ((GetBootMode() == BOOT_ON_S3_RESUME) && (GetPayloadId () == UEFI_PAYLOAD_ID_SIGNATURE)) {
-      EnableSmi ();
+      SmmBaseInfo = (SMMBASE_INFO *) (MmioRead32 (TO_MM_PCI_ADDRESS (0x00000000) + TSEG) & ~0xF);
+      S3RestoreRegData ((UINT8 *)SmmBaseInfo);
```

Comment:
How do we handle UpdateSmmInfo()?

Reply:
UpdateSmmInfo() is called only on Normal boot path.